### DISCUSSION TITLE
Agent-based mesoscopic traffic flow simulation model (Mesa + UXsim)

### DISCUSSION THREAD

**Discussion by EwoutH** [mesa/mesa-examples#167] [Show and tell]
I'm developing a hybrid Agent-based mesoscopic traffic flow simulation model, using [Mesa](https://github.com/projectmesa/mesa) and [UXsim](https://github.com/toruseo/UXsim). It's based on a real-world road network extracted and processed with [OSMnx](https://github.com/gboeing/osmnx). It's in early stages, but development can be followed here:

- https://github.com/EwoutH/urban-self-driving-effects/tree/main/model

Core features from Mesa used are the AgentSet and the experimental DEVS simulator.

Once it's in further stages I will describe it more extensively, and maybe open a PR to add (a simplified version) as an example. But for now a quick share for anyone interested to follow.

![self-driving-research-flow-Conceptual-models-v1 drawio (1)](https://github.com/user-attachments/assets/82486773-0ac6-4e64-b78d-ac5a2209e551)
_Early conceptual model_

**Comment by EwoutH** [comment]
Quick update: Model is running and has reached minimal-viable product status, with a latest interaction bug between Mesa and UXsim resolved (correct bookkeeping of time in both is important): https://github.com/EwoutH/urban-self-driving-effects/commit/2a702c36c916b88658edae16a8997fb6408e8ef5. It was an interesting case in using DEVS to schedule things for another simulation model.

The modal is simulating traffic in a ~half a million city during a full day, represented by 120.000 agents which each represent 5 people in the traffic model. UXsim is running with up to 50 thousand vehicles simultaneously. Each 900 seconds the model recalculates all travel times between all nodes on which allows Agents to take new mode choice decisions.

First traffic patterns are following expectations. Lot's of "magic numbers" of course, tomorrow is data collection, aggregation, plotting and validation day. Goal is to determine where to refine more.

Then it's taking a step back to the original research goals to see which questions we can exactly answer, what we need to be able to do that, which are unfeasible.

**Comment by PRIYANSHU2026** [comment]
Hi @EwoutH we successfully presented our  research paper titled "GIS-Based Urban Traffic Simulation by Mesa Framework" at the prestigious Springer International Conference CSEAI24 (2nd International Conference on Computing for Science, Engineering, and Artificial Intelligence)  The paper is currently in the process of being published and will be indexed in Scopus and Web of Science.

https://www.linkedin.com/feed/update/urn:li:activity:7283545533720190977/

[Paper_ID@AICCT2024[1].pptx  -  Read-Only.key.pdf](https://github.com/user-attachments/files/18386013/Paper_ID%40AICCT2024.1.pptx.-.Read-Only.key.pdf)

please review our PPT

**Reply by PRIYANSHU2026** [reply]
Hi @EwoutH, just a gentle reminder, I’m waiting for your feedback on the PPT. It’s been three days now. Please share your thoughts

**Reply by EwoutH** [reply]
Congratulations!

What do you want reviewed exactly with what purpose?

**Reply by PRIYANSHU2026** [reply]
I want your opinion and suggestions for improvement because this project is now moving forward to several hackathons. After publication, I will share the full-length paper. How I Integrated Mesa in this Research.

Thank you for motivating me to improve my research, communication, and core programming skills.

**Reply by EwoutH** [reply]
I think you could add some focus my defining specific research questions/goals, and then building the remainder of your presentation around that.

**Comment by PRIYANSHU2026** [comment]
**Hi sir @EwoutH,**

How are you? Sorry to disturb you — I just wanted to share some good news. I've been selected for **GSoC 2025** with **INCF (International Neuroinformatics Coordinating Facility)**!

It’s a dream come true, and I truly appreciate all your guidance and feedback along the way. I’ve been chosen for a large project based on LLMs.

Thank you so much for your Valuable Feedback!

Best regards,
**PRIYANSHU TIWARI**
